karapace/config.py

```diff
@@ -23,7 +23,7 @@ class Config(TypedDict):
     access_logs_debug: bool
     access_log_class: type | None
     advertised_hostname: str
-    advertised_port: int
+    advertised_port: int | None
```

Comment:
question: I believe `advertised_port` should not be ` | None` here, only in `ConfigDefaults`?

Reply:
I.e., the `Config` type should model a valid configuration, we should not have to do assertions when we have an instance of it, we should already know that it is a valid configuration that can run.

Reply:
Since we cannot run with `advertised_port=None`, we should eliminate that illegal state on the typing level.

Reply:
I didn't notice our linters would accept it that way. In my testing even if `def set_config_defaults(config: ConfigDefaults) -> Config:` would return dict with `advertised_port: None` it would pass mypy so feels a bit cheating. But maybe it is this way simpler anyway.

Reply:
@tvainika Gotcha, that's unfortunate, but my intention with the introduction of `TypedDict`s here was rather meant as stop gap to be able to move forward with typing, rather than as an end solution. For now, I guess the trust on the class is one-sided.

Reply:
I.e., the `Config` class makes a promise, and we trust that promise throughout the application, but nothing is actually verifying the promise. IMO we should use Pydantic for this whenever we get the chance.